packages/scanner/src/Scanner.ts

```diff
@@ -0,0 +1,225 @@
+import { isHex, isNumber, u8aToU8a } from '@polkadot/util';
+import { TypeRegistry, StorageKey, Vec, GenericExtrinsic } from '@polkadot/types';
+import Metadata from '@polkadot/metadata/Decorated';
+import { createTypeUnsafe } from '@polkadot/types/codec';
+import { EventRecord } from '@polkadot/types/interfaces/system';
+import { Observable, range, from, concat, of } from 'rxjs';
+import { switchMap, map, take, shareReplay, mergeMap, pairwise } from 'rxjs/operators';
+
+import {
+  BlockAt,
+  ScannerOptions,
+  RpcProvider,
+  Bytes,
+  BlockAtOptions,
+  Block,
+  BlockRaw,
+  Header,
+  Confirmation,
+  RuntimeVersion,
+  SubcribeOptions,
+  ChainInfo
+} from './types';
+
+class Scanner {
+  private rpcProvider: RpcProvider;
+  private chainInfo: Record<string, ChainInfo>;
+
+  constructor(options: ScannerOptions) {
+    this.rpcProvider = options.provider;
+    this.chainInfo = {};
+  }
+
+  private createMethodSubscribe<T>(methods: string[], ...params: any[]): Observable<T> {
+    const [updateType, subMethod, unsubMethod] = methods;
+
+    return new Observable<T>(observer => {
+      let subscriptionPromise: Promise<number | void> = Promise.resolve();
+      const errorHandler = (error: Error): void => {
+        observer.error(error);
+      };
+
+      try {
+        const update = (error?: Error, result?: any): void => {
+          if (error) {
+            // errorHandler(error)
+            return;
+          }
+          observer.next(result);
+        };
+
+        subscriptionPromise = this.rpcProvider
+          .subscribe(updateType, subMethod, params, update)
+          .catch(error => errorHandler(error));
+      } catch (error) {
+        errorHandler(error);
+      }
+
+      return (): void => {
+        subscriptionPromise.then(
+          (subscriptionId): Promise<boolean> =>
+            isNumber(subscriptionId)
+              ? this.rpcProvider.unsubscribe(updateType, unsubMethod, subscriptionId)
+              : Promise.resolve(false)
+        );
+      };
+    });
+  }
+
+  public async getBlockDetail(_blockAt?: BlockAtOptions): Promise<Block> {
+    const blockAt = await this.getBlockAt(_blockAt);
+    const blockRaw: BlockRaw = await this.rpcProvider.send('chain_getBlock', [blockAt.blockHash]);
```

Comment:
```suggestion
    const blockRaw: SignedBlock = await this.rpcProvider.send('chain_getBlock', [blockAt.blockHash]);
```

@aniiantt I think this needs to be `SignedBlock`
`import { SignedBlock } from '@polkadot/types/interfaces';`

Reply:
this.rpcProvider.send('chain_getBlock', [blockAt.blockHash]) return json. But SignedBlock is a complex object.